### ISSUE TITLE
[Rollup] Support range aggregations

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#33041]
I think we can support _some_ range aggregations on the existing data that we collect.  E.g. if the ranges specify are compatible with the bucketing intervals collected by the `date_histogram`/`histogram`, ranges will work as expected.

We could also implement new groupings specifically for the range aggregation.  It's a little less convenient, less efficient, and still has issues with bucket compatibility (there's no guarantee the ranges in the config match the ranges in the query).

**Comment by colings86** [comment]
I think supporting ranges at query time is a good idea. I think we should separate out the index time question into a different issue as that requires more work and I'm personally not sure we could implement it in a way thats easy to explain to the user

**Comment by polyfractal** [comment]
Agreed.  Once we get #32052 merged, I think implementing query-time ranges should be reasonably straightforward.  That PR does most of the heavy lifting wrt validating time intervals, so we can just piggyback on that and validate the ranges too.

**Comment by csoulios** [comment]
The team discussed this issue and it looks this idea is far from the current approach for rollups/downsampling.
